packages/core/src/anchor/memory/in-memory-anchor-service.ts

```diff
@@ -310,27 +309,44 @@ export class InMemoryAnchorService implements AnchorService, AnchorValidator {
     }
   }
 
+  async _storeRecord(record: Record<string, unknown>): Promise<CID> {
+    let timeout: any
+
+    const putPromise = this.#dispatcher._ipfs.dag.put(record).finally(() => {
```

Comment:
can use `dispatcher.storeCommit`, instead of reaching into its internal ipfs handle